### ISSUE TITLE
bug: no more cookies file in 1.1-beta5

### ISSUE THREAD

**Issue by yoannmoinet** [n1k0/casperjs#1448]
Hi,
I'm working from node with the `child_process.spawn` command like follow :

```javascript
var child = spawn(
    path.join(__dirname, '../node_modules/casperjs/bin/casperjs'),
    [
        path.join(__dirname, './bots/worker.js'),
        '--tasks=' + tasks,
        '--cwd=' + this.folder,
        '--cookies-file=' + this.cookie,
        '--web-security=no',
        '--log-file=' + this.log
    ]
);
```

All filenames are absolute.
I'm using most of the flags in my `worker.js`.

Everything works fine on `1.1-beta4` but there is no cookies-file on `1.1-beta5`.
So I guess that in this latest version flags may have a problem on Windows (couldn't test it on a *nix system yet).

Tested on **Windows 7** with **node 0.10.36** and **5.5.0**, with **PhantomJS 2.0**

I'd love to help so I'll try to find out how to build the Windows executable.

**Comment by istr** [comment]
Hm... Sorry to hear that. Maybe related to #1427 and the merged PRs  #1387, #1388, #1389.

> I'd love to help so I'll try to find out how to build the Windows executable.
Using mono this is as simple as:

```sh
make compile-dotNET
```

However, I am not sure how to install ```mcs``` and/or ```make``` on Windows in a sensible way...

**Comment by mepard** [comment]
I've compiled it on Windows by launching VS 2012 x86 Tools Command Prompt, then running:

```
cd \Users\Marc\Documents\GitHub\casperjs
"C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat"

csc.exe /out:bin\casperjs.exe src\casperjs.cs
```

**Comment by yoannmoinet** [comment]
Thanks @mepard, I'll try that.

**Comment by yoannmoinet** [comment]
By reversing this commit only c51d700 I have my cookies back.
So the issue might come from the regex checking introduced in #1384

**Comment by yoannmoinet** [comment]
Ok found it.

In `casperjs.cs` line 146, 
`string arg_name = arg_match.Groups[0].Captures[0].ToString();` should be 
`string arg_name = arg_match.Groups[1].Captures[0].ToString();`.

Otherwise the check is made on the complete string and not the correct captured group.

I'll submit a PR soon.

**Comment by mepard** [comment]
Sorry about that. You might beef up runtests.py to catch this (search for --whoops).

**Comment by yoannmoinet** [comment]
No problem, it was an easy catch.

Will check the test.

**Comment by istr** [comment]
@yoannmoinet Thanks for resolving this -- a PR is highly welcome.

**Comment by yoannmoinet** [comment]
No problem @istr. 
My branch is ready on [yoannmoinet/casperjs/fix-windows-engine-flags](https://github.com/yoannmoinet/casperjs/tree/fix-windows-engine-flags). 

I'll just look into the tests later on this evening (I've so much on my plate right now...) and Python isn't my language of predilection as it would be my first time with it.

And I'll submit the PR at that moment, or, if you prefer, I'll submit it right now, and add the testing later.

**Comment by istr** [comment]
A PR that actually fixes the bug would be sufficient for the moment, as this is a regression that would justify another "hotfix" / follow up release, maybe in conjunction with a fix for #1427.

We have [a problem testing windows anyway](https://github.com/n1k0/casperjs/issues/1427#issuecomment-180078441) so the bugfix would help a lot already.

Thank you.
